core/src/utils/mod.rs

```diff
@@ -194,23 +194,33 @@ pub fn log2_strict_usize(n: usize) -> usize {
     res as usize
 }
 
-pub fn par_for_each_row<P, F>(vec: &mut [F], num_cols: usize, processor: P)
-where
+pub fn par_for_each_row<P, F>(
+    vec: &mut [F],
+    num_rows_per_event: usize,
+    num_cols: usize,
```

Comment:
it's a bit confusing that these values aren't really used individually. should it just be one variable `num_items_per_row` or something? maybe the term row is misleading then though